network/src/protocols/network/mod.rs

```diff
@@ -349,3 +350,23 @@ impl<TMessage: Message> NetworkSender<TMessage> {
         Ok(res_msg)
     }
 }
+
+/// A simplified version of `NetworkSender` that doesn't use `ProtocolId` in the input
+/// It was already being implemented for every application, but is now standardized
+#[async_trait]
+pub trait ApplicationNetworkSender<TMessage: Send>: Clone {
+    fn send_to(&mut self, recipient: PeerId, message: TMessage) -> Result<(), NetworkError>;
+
+    fn send_to_many(
+        &mut self,
+        recipients: impl Iterator<Item = PeerId>,
```

Comment:
should we not provide a send_to_all or just a broadcast? how often are we calling this where recipients isn't all?

Reply:
`send_to_many` is a broadcast, which is currently being used by Mempool / State sync.

I haven't checked when recipients isn't all, but I'm assuming we broadcast either `upstream` or `downstream` which isn't all :D

Reply:
Okay, so originally this could be `send_to_all` instead of `send_to_many` because it's only used for consensus.  But, @zekun000 smartly has used it to send 2 different protocols to two groups of validators.  So, it can't be renamed for now.

Reply:
@zekun000 WHY!